binaries/cli/src/progress.rs

```diff
@@ -0,0 +1,433 @@
+//! Shared progress bar components for time-consuming CLI operations.
+//!
+//! This module provides a unified interface for displaying progress feedback
+//! across all dora CLI commands using the `indicatif` crate.
+//!
+//! # Design Principles
+//!
+//! - **Unified Style**: All progress bars share consistent styling and behavior
+//! - **Reusable Components**: Common patterns are encapsulated for easy reuse
+//! - **Multi-task Support**: Support for parallel operations with multiple progress bars
+//! - **Graceful Degradation**: Works in environments without TTY support
+//!
+//! # Usage Examples
+//!
+//! ## Simple Progress Bar
+//!
+//! ```rust,ignore
+//! use crate::progress::ProgressBar;
+//!
+//! let pb = ProgressBar::new_spinner("Building nodes...");
+//! // do work...
+//! pb.finish_with_message("Build complete!");
+//! ```
+//!
+//! ## Multi-task Progress
+//!
+//! ```rust,ignore
+//! use crate::progress::MultiProgress;
+//!
+//! let multi = MultiProgress::new();
+//! let pb1 = multi.add_task("Cloning repository...");
+//! let pb2 = multi.add_task("Building node...");
+//! // do work...
+//! pb1.finish();
+//! pb2.finish();
+//! ```
```

Comment:
Please also provide an example that shows an actual progress bar, instead of just spinners.